estimagic/optimization/kernel_transformations.py

```diff
@@ -14,42 +40,513 @@ def covariance_to_internal(external_values, constr):
+    jacobian can then be computed as
+
+    .. math::
+        J(f)(x) =
+        \frac{1}{x_m} \sum_{k=1}^{m-1} e_k e_k^\top -
+        \frac{1}{x_m^2}  [0, \dots, 0,
+            \left ( \begin{matrix} x_{1:m-1} \\ 0 \end{matrix} \right )
+        ]
+
+    Args:
+        external_values (np.ndarray): Array of probabilities; sums to one.
+
+    Returns:
+        deriv: The Jacobian matrix.
+
+    """
+    dim = len(external_values)
+
+    deriv = np.eye(dim) / external_values[-1]
+    deriv[:, -1] -= external_values / (external_values[-1] ** 2)
+    deriv[-1, -1] = 0
+
+    return deriv
+
+
 def probability_from_internal(internal_values, constr):
     """Reparametrize probability constrained parameters from internal."""
     return internal_values / internal_values.sum()
 
 
+def probability_from_internal_jacobian(internal_values, constr):
+    r"""Jacobian of ``probability_from_internal``.
+
+    Let :math:`x := \text{internal}`. The function ``probability_from_internal``
+    has the following structure
+
+    .. math::`f: \mathbb{R}^m \to \mathbb{R}^m, x \mapsto \frac{1}{x^\top 1} x`
+
+    where :math:`1` denotes a vector of all ones and :math:`I_m` the identity
+    matrix. The jacobian can be computed as
+
+    .. math::  J(f)(x) = \frac{1}{\sigma} I_m - \frac{1}{\sigma^2} 1 x^\top
+
+    Args:
+        internal_values (np.ndarray): Internal (positive) values.
+
+    Returns:
+        deriv: The Jacobian matrix.
+
+    """
+    dim = len(internal_values)
+
+    sigma = np.sum(internal_values)
+    left = np.eye(dim)
+    right = (np.ones((dim, dim)) * (internal_values / sigma)).T
+
+    deriv = (left - right) / sigma
+    return deriv
+
+
 def linear_to_internal(external_values, constr):
     """Reparametrize linear constraint to internal."""
     return constr["to_internal"] @ external_values
 
 
+def linear_to_internal_jacobian(external_values, constr):
+    return constr["to_internal"]
+
+
 def linear_from_internal(internal_values, constr):
     """Reparametrize linear constraint from internal."""
     return constr["from_internal"] @ internal_values
+
+
+def linear_from_internal_jacobian(internal_values, constr):
+    return constr["from_internal"]
+
+
+def _elimination_matrix(dim):
+    r"""Construct (row-wise) elimination matrix.
+
+    Let :math:`A` be a quadratic matrix. Let :math:`\text{vec}(A)` be the
+    column-wise vectorization of :math:`A`. Let :math:`\text{vech}(A)` be the
+    row-wise half-vectorization of :math:`A`. Then the corresponding elimination
+    matrix :math:`L` has the property
+
+    .. math::  L \text{vec}(A) = \text{vech}(A)
+
+    See the wiki entry https://tinyurl.com/yy4sdr43 for further information, but
+    note that here we are using :math:`\text{vech}` as the row-wise and not
+    column-wise half-vectorization.
+
+    Args:
+        dim (int): The dimension.
+
+    Returns:
+        eliminator (np.ndarray): The elimination matrix.
+
+    Example:
```

Comment:
I am not sure whether doctest wont pick up this test or that numpydoc does not format the example well but i think it must be plural.

```suggestion
    Examples:
```